library/src/scala/compiletime/package.scala

```diff
@@ -35,7 +35,7 @@ package object compiletime {
    /** Returns the string representation of interpolated elaborated code:
     *
     *  ```scala
-    *  inline def logged(p1: => Any) = {
+    *  inline def logged(inline p1: Any) = {
     *    val c = code"code: $p1"
```

Comment:
I guess it's a bit late to change this, but why is `code` a string interpolator instead of just a method from Any to String?

Reply:
It was to have something that produces the full statically known string that is fed to `error`.

Reply:
That use case is in the documentation of `error`

Reply:
So is the problem that if I wrote `s"My error: ${code(x)}"` the result wouldn't be a String literal and so couldn't be passed to `error`? Seems like it'd be nice to have logic to concatenate multiple String literals into one String literal in the typer, but it might be complicated to do.

Reply:
We do not constant fold `s""` in typer. If we would do that we could probably simplify it.

Reply:
Actually, now constant folding is good enough to do `"My error: " + code"$x")` and therefore could simplify it to `"My error: " + codeOf(x))`

Reply:
I would do that in another PR because it is a change of the API while this PR cleans up the library.

Reply:
Sounds good to me.

Reply:
See #10313

Reply:
That idea will simplify the API and implement a lot.